.github/workflows/parser-update.yaml

```diff
@@ -0,0 +1,107 @@
+name: Update parser version references
+
+on:
+  schedule:
+    - cron: "0 6 * * *"
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  update-parser:
+    if: github.repository_owner == 'redhat-best-practices-for-k8s'
+    name: Update parser tag and open PR
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: write
+      pull-requests: write
+    env:
+      SHELL: /bin/bash
+      PARSER_RELEASES_URL: https://api.github.com/repos/redhat-best-practices-for-k8s/parser/releases/latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+        with:
+          ref: main
+
+      - name: Set up Go
+        uses: actions/setup-go@44694675825211faa026b3c33043df3e48a5fa00 # v6.0.0
+        with:
+          go-version-file: go.mod
+
+      - name: Determine current and latest parser versions
+        id: versions
+        run: |
+          set -euo pipefail
+
+          CURRENT=$(jq -r '.parserTag' version.json)
+          echo "current=${CURRENT}" >> $GITHUB_OUTPUT
+
+          # Fetch latest release with retry and exponential backoff
+          AUTH_HEADER=""
+          if [[ -n "${GITHUB_TOKEN:-}" ]]; then
+            AUTH_HEADER="Authorization: Bearer ${GITHUB_TOKEN}"
+          fi
+
+          LATEST=""
+          for attempt in {1..5}; do
+            STATUS=$(curl -sS -H "Accept: application/vnd.github+json" ${AUTH_HEADER:+-H "$AUTH_HEADER"} -o /tmp/parser_latest.json -w "%{http_code}" "${PARSER_RELEASES_URL}" || true)
+            if [[ "${STATUS}" == "200" ]]; then
+              LATEST=$(jq -r .tag_name /tmp/parser_latest.json)
+            fi
+
+            if [[ -n "${LATEST}" && "${LATEST}" != "null" ]]; then
+              break
+            fi
+
+            SLEEP=$((2 ** (attempt-1)))
+            echo "Attempt ${attempt} failed (status=${STATUS}). Retrying in ${SLEEP}s..."
+            sleep "${SLEEP}"
+          done
```

Comment:
I think what you have works, but do you really want to maintain all of this custom logic? `gh` cli ships with the action runners, so I think something as simple as the below would work and you can drop most (all?) of the code.

`❯ gh release list -R redhat-best-practices-for-k8s/parser --json name,isLatest --jq '.[] | select(.isLatest)|.name'
v0.5.3
`